<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Lab 1: Getting started with R/RStudio</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PUBH 207B: Spring 2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="course_schedule.html">Course Schedule</a>
</li>
<li>
  <a href="install.html">Getting Started</a>
</li>
<li>
  <a href="about.html">Assignment Guidelines</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Lab Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Lab_1.html">Lab 1</a>
    </li>
    <li>
      <a href="Lab_2.html">Lab 2</a>
    </li>
    <li>
      <a href="Lab_3.html">Lab 3</a>
    </li>
    <li>
      <a href="Lab_4.html">Lab 4</a>
    </li>
    <li>
      <a href="Lab_5.html">Lab 5</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Lab 1: Getting started with R/RStudio</h1>

</div>


<p><br></p>
<p><strong>Jenny Wagner, PhD, MPH</strong> <br> Department of Public
Health <br> California State University, Sacramento <br> <br></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Lab 1 introduces you to the core functions of R/RStudio and covers
essential data cleaning techniques. Data cleaning is a critical skill to
have in carrying out public health research. It entails the preparation
of raw - and often messy and unstructured - data for analysis. While the
specific steps can vary by project, many common tasks — such as
modifying variable types, filtering data, and creating new variables —
are foundational to ensuring reliable and reproducible analyses.</p>
<p>Before you begin Lab 1, I recommend reading Chapter 2 (“The Very
Basics”) of <a
href="https://rstudio-education.github.io/hopr/basics.html">Hands-On
Programming with R</a>. This chapter provides an overview of the R
language and its use within RStudio. Some of this information has
already been covered on the Assignment Guidelines page; however, this
optional reading will provide more detail if you need additional
context.</p>
<p><br></p>
</div>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<p>After completing Lab 1, you will be able to:</p>
<ol style="list-style-type: decimal">
<li>Install and load R packages.</li>
<li>Set your working directory.</li>
<li>Import a dataset into RStudio.</li>
<li>Understand and use data documentation.</li>
<li>Filter a dataset.</li>
<li>Identify and modify variable types.</li>
<li>Rename a variable.</li>
<li>Create a new variable.</li>
<li>Export a dataset from RStudio.</li>
</ol>
<p><br></p>
</div>
<div id="tasks" class="section level1">
<h1>Tasks</h1>
<p>When you are ready to start Lab 1:</p>
<ol style="list-style-type: decimal">
<li>First create a new R Markdown file using the instructions on the
Assignment Guidelines page. Save this <code>.Rmd</code> file in a folder
dedicated to Lab 1 materials.</li>
<li>Next, download the dataset “frmgham2.csv” from Canvas and save it in
the same folder as your Lab 1 R Markdown file.</li>
<li>Finally, proceed to read through and carry out each of the tasks
detailed below. You will need to complete each task sequentially in
order to complete this assignment. In other words, you won’t be able to
complete task #3 without first completing task #2, etc.</li>
</ol>
<p><br></p>
<div id="install-and-load-r-packages" class="section level2">
<h2>1. Install and load R packages</h2>
<p>On the Assignment Guidelines page, you read through an example in
which R was used to add 1+1. R can be used in this way as a simple
calculator; however, most often we will use R functions (or “commands”)
to perform more complex analyses that would otherwise be extremely
burdensome to perform by hand. R functions allow you to perform tasks
such as data visualization, statistical modeling, or data cleaning,
without having to write all the code from scratch. For example, suppose
you want to know the mean (or average) of a set of numbers. Rather than
manually adding all of the values together and then dividing by the
number of values, we can simply use the <code>mean()</code> command in
R. These types of R functions do not exist in a vacuum but rather within
R “packages”, which are collections of pre-written functions, data, and
documentation that extend R’s capabilities far beyond a simple
calculator.</p>
<p>To use the functions contained within a particular package, that
package must be installed in R. You can install a package using the
<code>install.packages()</code> command. Once you have done this, the
package is saved on your computer, and you won’t need to install it
again unless you update or reinstall R.</p>
<p>In RStudio, we will work with many different packages over the course
of the semester. For now, start by installing the packages
<strong>knitr</strong>, <strong>rmarkdown</strong>, and
<strong>tidyverse</strong> by typing the following in your <strong>R
Console</strong> window (bottom-left) after <code>&gt;</code>:</p>
<pre><code>install.packages(&quot;knitr&quot;)
install.packages(&quot;rmarkdown&quot;)
install.packages(&quot;tidyverse&quot;)</code></pre>
<p>Remember that when we type code directly into the Console window,
that code is <em>not</em> saved for future use. This is exactly what we
want to do when installing packages for the first time. This is because,
as mentioned above, we only need to install a package once. However, to
use the functions from a package in a new R session, you must load the
package using the <code>library()</code> function. Loading a package
makes its functions available for use in your <em>current</em> R
session.</p>
<p>For example, to load the packages you installed a moment ago, create
a chunk in your R Markdown document as follows:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># load packages</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<p>After running this chunk, you will see a lot of “stuff” appear in the
Console window. For our purposes we can ignore all of this when loading
packages - that is, unless you see an error message, which would
indicate the package did not load properly.</p>
<p>After loading an R package, you can explore its functions by
accessing the package documentation. One way to do this is to use the
<code>help(package = "package_name")</code> command. For example:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># access &quot;tidyverse&quot; package documentation</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">help</span>(<span class="at">package =</span> <span class="st">&quot;tidyverse&quot;</span>)</span></code></pre></div>
<p>You should see some information about “tidyverse” appear in the Help
window (bottom-right), including links to a description file, user
guide, and other documentation.</p>
<p><br></p>
</div>
<div id="set-your-working-directory" class="section level2">
<h2>2. Set your working directory</h2>
<p>In RStudio, the <strong>working directory</strong> is the default
folder where R reads and saves files during your session. Think of it as
the “home base” for your current project. When you import a dataset or
save a file, R will look in this directory unless you specify otherwise.
When starting an assignment, you will always need to specify your
working directory. For example, for the current assignment (Lab 1), you
will want to set your working directory as the folder where you have
saved your Lab 1 R Markdown file and “frmgham2.csv” dataset.</p>
<p>You can set the working directory in a couple of ways:</p>
<ol style="list-style-type: decimal">
<li><p><em>Manually in RStudio:</em> Go to the “Session” menu at the top
of your screen, hover over “Set Working Directory”, and then click
“Choose Directory…” to navigate to your desired folder.</p></li>
<li><p><em>Using Code (Recommended):</em> Use the setwd() function to
specify the path to your directory. For example:</p></li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># set working directory</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/Volumes/GoogleDrive/My Drive/1_CSUS/2024-2025/Spring 2025/PUBH 207B/Website/PUBH-207B-S25&quot;</span>)</span></code></pre></div>
<p>You, of course, will need to insert your own file path between the
quotation marks. Not sure how to copy your file path? Try Googling “how
to copy file path on mac” (or on whatever operating system you are
using).</p>
<p>I recommend using code so that your working directory for a
particular assignment or project is documented in your R Markdown file.
This will be helpful when restarting R or reopening an assignment
because rather than searching through your files to set the working
directory manually, you will just need to rerun the chunk that already
contains the necessary file path.</p>
<p><br></p>
</div>
<div id="import-a-dataset-into-rstudio" class="section level2">
<h2>3. Import a dataset into RStudio</h2>
<p>To analyze data in RStudio, the first step is often to import a
dataset into your working environment. R supports various file types,
including CSV, Excel, and text files. The most common data file type you
will encounter is a CSV (Comma-Separated Values) file.</p>
<p>For Lab 1, you should have already downloaded the dataset called
“frmgham2.csv” and saved it in the same folder as your Lab 1 R Markdown
file. You should have also already set this folder as your working
directory.</p>
<p>To import a CSV file, use the <code>read.csv()</code> function,
specifying the name of your dataset within quotation marks as
follows:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># load dataset</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;frmgham2.csv&quot;</span>)</span></code></pre></div>
<p>Notice the line of code we used to bring in the dataset starts with
<code>data &lt;-</code>. This is telling R that we want to save the
“frmgham2.csv” dataset as an “object” in our Environment and call it
“data”. An R object is a named data structure in R that stores
information, such as data, functions, or results, allowing you to reuse
and manipulate it throughout your analysis. In this case, we want to be
able to access and manipulate the data contained within the
“frmgham2.csv” data file. This is why, after running this chunk, you
should see an R object called “data” in your Environment window
(top-right). When performing your own analyses, you can call R objects
whatever makes the most sense to you. A couple of quick observations
about the “data” object now in our Environment:</p>
<ul>
<li>The spreadsheet icon on the far right indicates that this object is
a “data frame” (i.e., a spreadsheet with rows and columns). For now,
know that there are different types of R objects we can store in the
Environment. Data frames are the most common type (but not the only
type) of R object we will work with.</li>
<li>The object is shown to contain “11627 obs. of 39 variables” - this
means that the dataset contains 11627 observations (rows) and 39
variables (columns).</li>
</ul>
<p>In the Environment window, click on the “data” object you have
created. You can also type <code>View(data)</code> into the Console for
the same result. You should see the dataset open in a new tab within
RStudio. It should essentially look like a spreadsheet, with variable
names across the top and each observation in a row.</p>
<p><em>Note:</em> For Excel files, you’ll need to install and load the
“readxl” package and use the <code>read_excel()</code> function instead
of <code>read.csv()</code>.</p>
<p><br></p>
</div>
<div id="understand-and-use-data-documentation" class="section level2">
<h2>4. Understand and use data documentation</h2>
<p>We’ve loaded our dataset into RStudio and now we’re ready to analyze
it, right? Not quite! Before you begin any kind of analysis (and really
before you even start downloading any data), it is important to have a
clear understanding of the dataset’s contents. This includes knowing
what each variable represents, how the data were collected, and whether
there are any limitations or nuances you need to account for. Without
this understanding, you risk misinterpreting results or making errors in
your analysis.</p>
<p>Data documentation, often called a <strong>codebook</strong> or
<strong>data dictionary</strong>, is a resource that explains:</p>
<ul>
<li>The variables in your dataset, including names, definitions, and
units of measurement.</li>
<li>The structure of the data (e.g., rows as individuals, columns as
variables).</li>
<li>Any codes or categories used (e.g., “1 = Male, 2 = Female”).</li>
<li>The context in which the data was collected (e.g., survey methods,
population characteristics).</li>
</ul>
<p>You will use data documentation at various stages of a project:</p>
<ul>
<li><em>Before analysis</em> - Familiarize yourself with the dataset by
reviewing the documentation to understand its structure and
variables.</li>
<li><em>During analysis</em> - Refer back to the documentation when you
encounter unfamiliar variables or need clarification.</li>
<li><em>After analysis</em> - Use the documentation to accurately
describe your methods and findings.</li>
</ul>
<p>For several assignments in this course, we will be using the
“frmgham2.csv” dataset you’ve loaded into your current R session. This
dataset is from the <strong>Framingham Heart Study</strong>, which is a
prospective cohort study carried out over several decades to study
causes and risk factors for cardiovascular disease. <strong>Before
moving forward, read over the first 7 pages of the data
documentation</strong>, linked on the Lab 1 assignment page in
Canvas.</p>
<p>Beginning on page 2 of the data documentation, you will see a table
that defines each of the variables included in the dataset, including
the variable name, a brief description, the units of the variable, and
the range of values (or, instead, the count of observations, labeled
with “n”). Take a moment to examine the dataset in RStudio in the
context of this codebook. You should see that each column in the dataset
has a corresponding row in the codebook that defines that variable.</p>
<p>Using the data documentation, answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>What does the variable called “RANDID” represent?</li>
<li>What does the variable called “PERIOD” represent? What are the
possible values of “PERIOD” and what do they represent?</li>
</ol>
<p><br></p>
</div>
<div id="filter-a-dataset" class="section level2">
<h2>5. Filter a dataset</h2>
<p><img src="filter.png" width="100%" /></p>
<p>One key benefit of working with a dataset in RStudio is that any data
manipulation you perform — such as filtering rows, creating new
variables, or transforming data — occurs within R and does
<strong>not</strong> alter the original data file. This ensures the raw
data remain unchanged, allowing you to experiment freely while
maintaining the integrity of your original dataset.</p>
<p>For example, using the dataset you’ve loaded from the Framingham
Heart Study, suppose we want to restrict the data to the
<em>baseline</em> sample for some exploratory analysis. As you have
likely already observed, the original dataset contains multiple
observations (up to three) for each participant, where each of these
observations corresponds to a different period of time (specified by the
“PERIOD” variable). The baseline observation (where PERIOD = 1) is the
initial data collected when a participant entered the study. We can
restrict the data to the baseline sample using the <code>filter()</code>
command as follows:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># filter data to include only baseline sample</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>data_period1 <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(PERIOD <span class="sc">==</span> <span class="dv">1</span>)</span></code></pre></div>
<p>First, understand that “filtering” data basically means that we are
selecting certain rows based on condition. There are a few things
happening in the above code chunk, so let’s break them down:</p>
<ul>
<li><code>filter(PERIOD == 1)</code>: Applies the <code>filter()</code>
function to select rows where the variable PERIOD is equal to 1. Only
rows meeting this condition are included in the resulting dataset.</li>
<li><code>data %&gt;%</code>: The <code>%&gt;%</code> is called a “pipe
operator” and comes from the “tidyverse” package. There are always
multiple ways the same task can be accomplished in R, and this operator
is essentially used to help us write our code in a logical and efficient
way. Here, the pipe operator is used to pass the dataset we called
“data” as <em>input</em> to the next function. In other words, the pipe
operator (%&gt;%) is like saying “and then.” It takes whatever is on the
left side — in this case, the dataset we called “data” — and hands it
over to the next step. So, you start with your data, <em>and then</em>
you tell R to filter it to only include rows where the PERIOD variable
is equal to 1. It’s like giving R a to-do list: ‘Start with this data
<em>and then</em> filter it for me.’ We will use the pipe operator a lot
throughout the course.</li>
<li><code>data_period1 &lt;-</code>: Creates a new object called
“data_period1” to store the filtered dataset. The <code>&lt;-</code>
operator is like an arrow - it assigns the result of the operation
(everything on the right of the arrow) to this new object. In other
words, we are taking our original dataset (called “data”), filtering it
to only include observations where PERIOD is equal to 1, and then saving
this filtered dataset as a new data object called “data_period1”. Giving
a new name to the resulting dataset is important because it allows us to
keep the original data unchanged while working with a modified version.
This way, the original data remains intact for reference or other
analyses, and the new name makes it clear what the filtered or modified
dataset represents. For example, naming the filtered dataset
“data_period1” helps us quickly identify that it contains only the rows
where PERIOD is equal to 1, making our work more organized and easier to
understand later.</li>
</ul>
<p>Once you run the code chunk above, you should see the new R object
called “data_period1” in your Environment (top-right window). Notice
that this object contains 4434 observations and 39 variables. Click on
the “data_period1” object to view it in a new tab. You should see that
all 39 variables in our original dataset have been retained. But, when
you scroll over to the PERIOD variable, all of the observations show a
value of 1.</p>
<p>We will continue to work with this modified dataset for the remainder
of the assignment.</p>
<p><br></p>
</div>
<div id="identify-and-modify-variable-types" class="section level2">
<h2>6. Identify and modify variable types</h2>
<p>Identifying and modifying variable types in R is crucial because the
type of a variable (e.g., numeric, character, factor) determines how R
processes and analyzes the data. A variable may be most appropriately
assigned one type—such as categorical—but R may initially interpret it
differently, such as numeric, depending on how the data is formatted or
imported. For example, a variable representing survey responses like “1
= Yes” and “2 = No” might be read as numeric, but treating it as a
categorical (factor) variable is more appropriate for analysis. Ensuring
variables have the correct type helps prevent errors, ensures accurate
analyses, and allows you to use the right functions effectively. By
explicitly modifying variable types when needed, you can make sure R
treats your data correctly for your analysis.</p>
<p>For example purposes, let’s identify the variable type for a few of
the variables in our dataset - specifically, sex, age, BMI (body mass
index), and attained education. We will do this using the “class”
function as follows:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># identify variable types</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>SEX) <span class="co"># sex</span></span></code></pre></div>
<pre><code>## [1] &quot;integer&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>AGE) <span class="co"># age</span></span></code></pre></div>
<pre><code>## [1] &quot;integer&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>BMI) <span class="co"># BMI</span></span></code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>educ) <span class="co"># attained education</span></span></code></pre></div>
<pre><code>## [1] &quot;integer&quot;</code></pre>
<p>When we run the above code chunk, the results will appear in the
Console (bottom-left window). You should see that <code>SEX</code> is an
“integer” variable, <code>AGE</code> is an “integer” variable,
<code>BMI</code> is “numeric”, and <code>educ</code> (attained
education) is an “integer” variable. Integer and numeric variables both
represent numbers - the difference is that integer variables are
discrete (the values must be whole numbers) whereas numeric variables
can take on any value (including decimals). R interprets variables as
integer or numeric by default when importing data because it relies on
the format of the raw data and doesn’t automatically assign “context” or
“meaning” to the values. For example, R doesn’t know that for “SEX”, 1
means male and 2 means female. It only sees the values “1” and “2”.
Using what we have learned in class, let’s consider the most appropriate
variable type for each of these four variables:</p>
<ul>
<li><strong>SEX:</strong> Represents categorical information about
biological sex, coded numerically (e.g., 1 = Male, 2 = Female). Although
represented as integers, the numbers are codes for categories rather
than quantities. This means that this variable would most appropriately
be coded as a “factor” variable. Treating it as a factor ensures R
correctly interprets and analyzes it as categorical data, allowing for
grouping and comparisons.</li>
<li><strong>AGE:</strong> Represents a person’s age, measured in whole
years. Age is a continuous, quantitative variable. Since age is being
measured in whole years, this variable is correctly coded as an
“integer” variable. Treating this variable as “numeric” would also be
appropriate.</li>
<li><strong>BMI:</strong> Represents body mass index, a continuous
measure derived from a person’s height and weight. BMI is inherently a
continuous, quantitative variable that can take decimal values, so the
default “numeric” type is appropriate for calculations and
analyses.</li>
<li><strong>educ:</strong> Represents a person’s level of attained
education, coded numerically (1=0-11 years; 2=High School Diploma, GED;
3=Some College, Vocational School; 4=College (BS, BA) degree or more).
Similar to sex, the numbers represent categories rather than quantities.
As such, this variable would most appropriately be coded as a “factor”
variable. Treating it as a factor ensures that R interprets it as
categorical data, making analyses like frequency counts or group
comparisons more meaningful.</li>
</ul>
<p>Given the above explanations, we’ve can conclude that the AGE and BMI
variables are already appropriately coded as “integer” and “numeric”,
respectively. SEX and educ, on the other hand, would be more
appropriately coded as factor (or “categorical”) variables given the
values represent categories. As we learned in class, these factor
variables can be further categorized into two types - ordered and
unordered. For example, sex represents categories with no inherent order
or ranking. In our dataset, sex can take on two values - 1 (male) or 2
(female). There is no logical sequence between these categories — they
are distinct but equal. As such, “SEX” should be treated as an unordered
factor variable in any analyses. Conversely, the values associated with
attained education represent categories with a natural hierarchy or
ranking (e.g., high school &lt; some college &lt; college). The levels
have a logical progression, with one level ranked above or below
another. As such, “educ” should be treated as an ordered factor variable
in any analyses. Ordered factors are treated as ordinal variables in
analyses, allowing you to consider their ranking in models or
visualizations.</p>
<p>To ensure these variables are utilized and interpreted correctly in
our analyses, we need to convert the variables to their most appropriate
types. We can do this using the <code>factor()</code> command as
follows:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># convert sex to unordered factor variable</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>data_period1<span class="sc">$</span>SEX <span class="ot">&lt;-</span> <span class="fu">factor</span>(data_period1<span class="sc">$</span>SEX, </span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>                           <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>), </span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>                           <span class="at">ordered=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>This code converts the SEX column of the data_period1 dataset into a
categorical variable (factor), specifying that it has two levels (“1”
and “2”) and that these levels are unordered. This ensures that R treats
the variable as distinct categories, making it suitable for categorical
analyses. Let’s break down the code one piece at a time:</p>
<ul>
<li><code>data_period1$SEX &lt;-</code>: This part assigns the result of
the <code>factor()</code> function to the <code>SEX</code> column of the
<code>data_period1</code> dataset. The <code>&lt;-</code> symbol means
“assign this value to,” so we are updating the <code>SEX</code> column
with a new version where it is converted into a factor. The
<code>$</code> operator here is used to access a specific column within
a dataset. The part before the <code>$</code> identifies the dataset to
use (in this case, <code>data_period1</code>), and the part after the
<code>$</code> specifies the column being modified (in this case,
<code>SEX</code>).</li>
<li><code>factor(data_period1$SEX, ... )</code>: The factor() function
is used to convert the SEX variable into a factor. Factors are used for
categorical data, allowing R to treat the variable as distinct
categories rather than as numeric values.</li>
<li><code>data_period1$SEX</code>: Refers to the SEX column in the
data_period1 dataset. This is the variable being transformed into a
factor.</li>
<li><code>levels=c("1", "2")</code>: Specifies the possible values (or
levels) for the factor. In this case, the levels are “1” and “2”, which
might correspond to categories like “Male” and “Female”. Defining levels
ensures R knows all valid categories for the factor.</li>
<li><code>ordered=FALSE</code>: Indicates that the factor is not
ordered—the levels do not have a natural hierarchy or ranking. For
example, “Male” and “Female” are categories with no logical order, so
ordered=FALSE is appropriate here.</li>
</ul>
<p><br></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># convert attained education to ordered factor variable</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>data_period1<span class="sc">$</span>educ <span class="ot">&lt;-</span> <span class="fu">factor</span>(data_period1<span class="sc">$</span>educ, </span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>                            <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;3&quot;</span>, <span class="st">&quot;4&quot;</span>), </span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>                            <span class="at">ordered =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>This code converts the <code>educ</code> column in the
<code>data_period1</code> dataset into an ordered factor with four
levels (“1”, “2”, “3”, “4”). The <code>ordered=TRUE</code> argument
tells R that the levels have a hierarchy, ensuring R interprets the
variable as ordinal, which is appropriate for analyses involving ranked
data, like educational attainment. Let’s break down the code one piece
at a time:</p>
<ul>
<li><code>data_period1$educ &lt;-</code>: This assigns the result of the
<code>factor()</code> function to the <code>educ</code> column of the
<code>data_period1</code> dataset. The &lt;- symbol means “assign this
value to,” so we are updating the educ column with a new version where
it is converted into a factor.</li>
<li><code>factor(... )</code>: As we saw above for <code>SEX</code>, the
factor() function is used to convert the <code>educ</code> variable into
a factor.</li>
<li><code>data_period1$educ</code>: Refers to the educ column within the
data_period1 dataset. This is the variable being transformed into an
ordered factor.</li>
<li><code>levels=c("1", "2", "3", "4")</code>: Specifies the possible
values (or levels) for the factor. In this case, the levels are “1”,
“2”, “3”, and “4”. These correspond to the categories shown in the data
documentation. Defining levels explicitly ensures R knows the valid
categories and their sequence.</li>
<li><code>ordered=TRUE</code>: Indicates that the factor is ordered,
meaning the levels have a natural hierarchy or ranking. For example, “1”
(e.g., “0-11 years”) is considered lower than “4” (e.g., “College (BS,
BA) degree or more”), and R will treat this as an ordinal variable,
allowing for comparisons that account for this order.</li>
</ul>
<p>When we assign the modified variable back to the <em>same name as the
original</em> (e.g., <code>data_period1$educ &lt;-</code>), R overwrites
the original variable with the new version. In both of our examples
here, the <code>SEX</code> and <code>educ</code> columns - originally
integer variables - are replaced with factor versions of the same
data.</p>
<p>This means the <code>SEX</code> and <code>educ</code> variables in
the <code>data_period1</code> dataset will no longer be treated as
integers. Instead, they will now be treated as factors, with its levels
and order defined by the factor() function. Overwriting these variables
ensures that R uses the updated type in future analyses, eliminating the
need to create a separate variable or keep track of multiple versions.
However, if you want to preserve the original version, you can assign
the modified variable a new name instead.</p>
<p>Now that we have modified the variable types for sex and attained
education, let’s check them using the <code>class()</code> function once
again:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># recheck variable types for modified variables</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>SEX) <span class="co"># sex</span></span></code></pre></div>
<pre><code>## [1] &quot;factor&quot;</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">class</span>(data_period1<span class="sc">$</span>educ) <span class="co"># attained education</span></span></code></pre></div>
<pre><code>## [1] &quot;ordered&quot; &quot;factor&quot;</code></pre>
<p>You should see that both <code>SEX</code> and <code>educ</code> are
now coded as “factor” variables. Hooray!</p>
<p><br></p>
</div>
<div id="rename-a-variable" class="section level2">
<h2>7. Rename a variable</h2>
<p>Renaming a variable during an analysis can make your work clearer,
more organized, and easier to understand. Here are a few reasons why you
might want to rename a variable:</p>
<ul>
<li>Original variable names may be unclear or cryptic (e.g., “var1”),
and renaming them to something descriptive (e.g., “Age”) makes the
dataset easier to interpret.</li>
<li>Renaming variables can help you maintain uniform naming conventions,
which is especially helpful when combining datasets or working
collaboratively.</li>
<li>Descriptive variable names are more intuitive for presentations,
reports, or sharing results with others who may not know the dataset as
well as you do. *Clear variable names reduce the risk of confusion or
mistakes in your analysis, especially when working with many
variables.</li>
</ul>
<p>In general, the variables included in the Framingham Heart Study
dataset are already named in fairly intuitive ways. The one
inconsistency is the <code>educ</code> variable is all lowercase,
whereas all other variables are uppercase. As long as we specify the
variable name correctly in our analyses, we wouldn’t necessarily
<em>need</em> to correct this inconsistency. But, for example purposes,
let’s look at how we could rename this variable using the
<code>rename()</code> function:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># rename &quot;educ&quot; to &quot;EDUC&quot;</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>data_period1 <span class="ot">&lt;-</span> data_period1 <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">EDUC =</span> educ)</span></code></pre></div>
<p>This code renames the <code>educ</code> variable in the
<code>data_period1</code> dataset to <code>EDUC</code>. The updated
dataset is then reassigned to <code>data_period1</code>, which ensures
the change is saved. Using <code>rename()</code> is useful for making
variable names more descriptive, consistent, or aligned with preferred
conventions. Let’s break down this line of code:</p>
<ul>
<li><code>rename(EDUC = educ)</code>: The <code>rename()</code> function
is used to change the name of a variable. Note that the sequence here is
“new name” = “current name”. <code>EDUC</code> is the new name for the
variable; <code>educ</code> is the current name of the variable that is
being renamed.</li>
<li><code>data_period1 %&gt;%</code>: As we saw previously, this uses
the “pipe operator” (<code>%&gt;%</code>) to pass the
<code>data_period1</code> dataset as input to the <code>rename()</code>
function. Remember, this basically tells R, ‘use the data in
<code>data_period1</code> <em>and then</em> rename educ to EDUC.’</li>
<li><code>data_period1 &lt;-</code>: Assigns the result of the operation
back to the <code>data_period1</code> dataset. This means the updated
dataset with the renamed variable will overwrite the original
version.</li>
</ul>
<p><br></p>
</div>
<div id="create-a-new-variable" class="section level2">
<h2>8. Create a new variable</h2>
<p><img src="mutate.png" width="100%" /></p>
<p>Creating new variables is an essential and common step in data
cleaning to prepare for analysis. Here are some situations where you
might want to create a new variable:</p>
<ul>
<li>When you need to transform or combine existing variables to make the
dataset ready for analysis (e.g., converting “height” and “weight” into
“BMI”).</li>
<li>To make analysis easier by summarizing data or recoding complex
variables into simpler categories (e.g., grouping age into age
ranges).</li>
<li>When you need new values derived from existing ones (e.g.,
calculating the difference between two dates to get “time
elapsed”).</li>
<li>Some analyses require variables to be recoded or reformatted to meet
specific requirements (e.g., creating binary variables for logistic
regression - more on this later in the course).</li>
</ul>
<p>In summary, creating new variables ensures your dataset accurately
reflects the questions you’re trying to answer and prepares it for
meaningful analysis. It’s a critical step in tailoring raw data to fit
the goals of your research.</p>
<p>Let’s consider an example using the Framingham Heart Study dataset.
Suppose we want to create a new variable that is a binary indicator of
“obese” status, defined as body mass index (BMI) greater than 30. We can
create a new variable, which we will call <code>OBESE</code>, using the
<code>mutate()</code> and <code>ifelse()</code> functions as
follows:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># create a new variable for obese status</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>data_period1 <span class="ot">&lt;-</span> data_period1 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">OBESE =</span> <span class="fu">ifelse</span>(BMI<span class="sc">&gt;</span><span class="dv">30</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span></code></pre></div>
<p>This code creates a new variable called <code>OBESE</code> in the
<code>data_period1</code> dataset. The variable is binary: it assigns a
value of 1 if the BMI is greater than 30 (indicating obesity) and 0
otherwise. The updated dataset, with the new variable, is reassigned to
<code>data_period1</code>. Let’s break down each piece of this code:</p>
<ul>
<li><code>mutate(OBESE = ... )</code>: The mutate() function is used to
create or modify variables in a dataset. <code>OBESE =</code>: Defines
the name of the new variable being created. In this case, the new
variable is called OBESE.</li>
<li><code>ifelse(BMI &gt; 30, 1, 0)</code>: The ifelse() function
evaluates a condition and assigns a value based on whether the condition
is true or false. In this case, BMI&gt;30 is the condition being
checked. For each row, it evaluates whether the BMI value is greater
than 30. If BMI&gt;30 (i.e., the condition is true), the value assigned
to the new variable is 1. If BMI &lt;= 30 (i.e., the condition is
false), the value assigned to the new variable is 0.</li>
<li><code>data_period1 %&gt;%</code>: Uses the pipe operator (%&gt;%) to
pass the data_period1 dataset into the mutate() function. This allows
you to modify the dataset by adding or changing variables.</li>
<li><code>data_period1 &lt;-</code>: Assigns the result of the operation
back to the <code>data_period1</code> dataset. This overwrites the
original dataset with the updated version that includes the new variable
OBESE.</li>
</ul>
<p>After running the code chunk above, the <code>data_period1</code>
object should now contain 40 variables (it previously had 39). This is
because we added a new variable, OBESE, to the dataset. If you click on
the <code>data_period1</code> object in the Environment tab to view the
dataset, you’ll see the new <code>OBESE</code> variable in the far-right
column. Each row will have a value of either 1 (indicating BMI&gt;30) or
0 (indicating BMI≤30), based on the calculation you just performed. This
confirms that the new variable was successfully created and added to the
dataset.</p>
<p><br></p>
</div>
<div id="export-a-dataset-from-rstudio" class="section level2">
<h2>9. Export a dataset from RStudio</h2>
<p>After completing data cleaning tasks, exporting the modified dataset
as a new CSV file can be very useful. This step allows you to save your
cleaned and prepared data for later use so that you don’t need to repeat
the data cleaning process every time you want to analyze the data.</p>
<p>Saving the cleaned dataset as a new file is particularly helpful
because:</p>
<ul>
<li>It keeps your cleaned data separate from the original raw data,
preserving the original dataset for reference or auditing.</li>
<li>You can directly load the cleaned dataset in future R sessions,
saving time and avoiding repetitive cleaning steps.</li>
<li>A cleaned and well-structured dataset can be easily shared with
colleagues or collaborators.</li>
<li>Exporting the cleaned dataset documents the result of your data
cleaning process, which is important for transparency and
reproducibility in public health research.</li>
</ul>
<p>To review, we have made the following modifications up to this point:
(1) filtered the data to retain only the baseline sample (period 1); (2)
converted the sex and attained education variables to factors; (3)
renamed the attained education variable for consistency with the format
of the other variables; and (4) created a new binary variable for obese
status. Suppose you want to export this modified dataset you have
created. We can use the <code>write.csv()</code> function as
follows:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># exports modified dataset as CSV</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="fu">write.csv</span>(data_period1, <span class="st">&quot;frmgham2_p1.csv&quot;</span>)</span></code></pre></div>
<p>The <code>write.csv()</code> function in R is used to export a
dataset into a CSV (Comma-Separated Values) file format. In this code,
the dataset <code>data_period1</code> (which contains all of the data
modifications we have made) is being saved as a file named
<code>frmgham2_p1.csv</code>. The filename is specified in quotes and
includes the <code>.csv</code> extension, and the file will be saved in
your <strong>current working directory</strong> unless a specific file
path is provided. This allows you to save your cleaned and modified
dataset for future use so that you don’t need to repeat the data
cleaning process. Exporting data in this way is also useful for sharing
your dataset with collaborators or using it in other software tools.</p>
<p><br></p>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>In Lab 1, you’ve gained hands-on experience with essential data
cleaning and preparation tasks in R. You’ve learned how to install and
load R packages, set your working directory, and import a dataset into
RStudio. Using data documentation, you explored the dataset’s structure
and variables, then applied key cleaning techniques: filtering the
dataset, identifying and modifying variable types, renaming a variable,
and creating a new variable. Finally, you exported the cleaned dataset
as a CSV file to ensure it is ready for future use or sharing. These
foundational skills are critical for preparing real-world public health
datasets for analysis.</p>
<p>You will continue to apply these skills in assignments throughout the
course as we build on this foundation. Lab 1 has provided a brief
introduction to important data cleaning techniques and coding practices,
which you will refine and expand upon as we progress through the
course.</p>
<p>When you are ready, please submit the following to the Lab 1
assignment page on Canvas:</p>
<ol style="list-style-type: decimal">
<li>An R Markdown document, which has a <code>.Rmd</code> extension</li>
<li>A knitted <code>.html</code> file</li>
</ol>
<p>Please reach out to me at <a href="mailto:jenny.wagner@csus.edu"
class="email">jenny.wagner@csus.edu</a> if you have any questions. See
you in class!</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
